azure/durable_functions/tasks/task_utilities.py

```diff
@@ -21,15 +21,66 @@ def parse_history_event(directive_result):
 
     # We provide the ability to deserialize custom objects, because the output of this
     # will be passed directly to the orchestrator as the output of some activity
-    if event_type == HistoryEventType.EVENT_RAISED:
-        return json.loads(directive_result.Input, object_hook=_deserialize_custom_object)
+    # TODO: why do we have this chain of equivalent if-statements?
     if event_type == HistoryEventType.SUB_ORCHESTRATION_INSTANCE_COMPLETED:
         return json.loads(directive_result.Result, object_hook=_deserialize_custom_object)
     if event_type == HistoryEventType.TASK_COMPLETED:
         return json.loads(directive_result.Result, object_hook=_deserialize_custom_object)
+    if event_type == HistoryEventType.EVENT_RAISED:
+        # TODO: why is the result here on input?
+        return json.loads(directive_result.Input, object_hook=_deserialize_custom_object)
```

Comment:
This does seem a little weird...

Reply:
Yep, it's strange....

Reply:
In any case, this is where that payload resides, so I suppose we should keep it as-is.